### PR TITLE
300 update get lobbylobbyidstatus to return game data

### PR DESCRIPTION
**Pull request by flamion**

### CODE REVIEW

**Review by TR1NITROTOLUOL — DISMISSED** [review]
Nice little adjustments to old tests and some code cleanup. I like those changes. The rest seems fine to me. Not necessarily a change request, but have you considered to add negative cases for the new tests e.g. trying to get the visualisation data before a game is finished? All in all i would approve, as it is now.

**Comment by flamion** [comment]
> Nice little adjustments to old tests and some code cleanup. I like those changes. The rest seems fine to me. Not necessarily a change request, but have you considered to add negative cases for the new tests e.g. trying to get the visualisation data before a game is finished? All in all i would approve, as it is now.

You're right, I'll add a negative test

**Review by TR1NITROTOLUOL — DISMISSED** [review]
No further remarks from me. Looks good to me.

File: backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt [hunk 1]
```diff
@@ -22,6 +23,55 @@ class ShorkInterpreterControllerV1IT : AbstractControllerTest() {
         }
     }
 
+    @Serializable data class Program(val code: String)
+
+    // Workaround for content negotiation plugin bullshit
```
Comment (nikkischnelle) [inline backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt:28]: I don't know if we want this comment in main :sweat_smile:

File: backend/api/src/test/kotlin/software/shonk/application/service/ShorkServiceTest.kt [hunk 2]
```diff
@@ -290,4 +320,30 @@ class ShorkServiceTest {
             )
         )
     }
+
+    @Test
+    fun `game visualization data works`() {
```
Comment (nikkischnelle) [inline backend/api/src/test/kotlin/software/shonk/application/service/ShorkServiceTest.kt:325]: This doesn't really test if it "works" just that there is data. The Test name should reflect that imo

**Review by nikkischnelle — CHANGES_REQUESTED** [review]

**Review by nikkischnelle — APPROVED** [review]
lgtm

catspin

**Pull request merged**